github/copilot_test.go

```diff
@@ -0,0 +1,649 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// Test invalid JSON responses, vlaid responses are covered in the other tests
+func TestCopilotSeatDetails_UnmarshalJSON(t *testing.T) {
+	tests := map[string]struct {
+		data    string
+		want    *CopilotSeatDetails
+		wantErr bool
+	}{
```

Comment:
This could be:

```go
tests := map[string]struct {
  name string
  data string
} {
  {
    name: "invalid JSON",
    data: `{`,
  },
...
}
  
```

Two things changed here:
- Make it a slice instead of a map for defined order when testing.
- Because this only tests invalid, the "want" and "wantErr" fields are always the same.

Reply:
I'll defer to your judgement on want and wantErr. We may or may not need them in the future, and it doesn't hurt anything to keep them around.